src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -108,52 +105,46 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig, gdDAO: Datap
      - Upload all the necessary initialization files to the bucket
      - Create the cluster in the google project
    Currently, the bucketPath of the clusterRequest is not used - it will be used later as a place to store notebook results */
-  private[service] def createGoogleCluster(googleProject: GoogleProject, clusterName: String, clusterRequest: ClusterRequest)(implicit executionContext: ExecutionContext): Future[ClusterResponse] = {
-    val bucketName = s"${clusterName}-${UUID.randomUUID.toString}"
+  private[service] def createGoogleCluster(googleProject: GoogleProject, clusterName: ClusterName, clusterRequest: ClusterRequest)(implicit executionContext: ExecutionContext): Future[ClusterResponse] = {
+    val bucketName = GoogleBucket(s"${clusterName.string}-${UUID.randomUUID.toString}")
     for {
       // Validate that the Jupyter extension URI is a valid URI and references a real GCS object
       _ <- validateJupyterExtensionUri(googleProject, clusterRequest.jupyterExtensionUri)
       // Create the firewall rule in the google project if it doesn't already exist, so we can access the cluster
       _ <- gdDAO.updateFirewallRule(googleProject)
       // Create the bucket in leo's google bucket and populate with initialization files
-      _ <- initializeBucket(dataprocConfig.leoGoogleBucket, clusterName, bucketName, clusterRequest)
+      _ <- initializeBucket(GoogleProject(dataprocConfig.leoGoogleBucket), clusterName, bucketName, clusterRequest)
       // Once the bucket is ready, build the cluster
       clusterResponse <- gdDAO.createCluster(googleProject, clusterName, clusterRequest, bucketName)
     } yield {
       clusterResponse
     }
   }
 
-  private[service] def validateJupyterExtensionUri(googleProject: GoogleProject, gcsUriOpt: Option[String])(implicit executionContext: ExecutionContext): Future[Unit] = {
+  private[service] def validateJupyterExtensionUri(googleProject: GoogleProject, gcsUriOpt: Option[GoogleBucketUri])(implicit executionContext: ExecutionContext): Future[Unit] = {
     gcsUriOpt match {
       case None => Future.successful(())
       case Some(gcsUri) =>
-        if (gcsUri.length > bucketPathMaxLength) {
+        if (gcsUri.string.length > bucketPathMaxLength) {
           throw JupyterExtensionException(gcsUri)
         }
-
-        val parsedUri = try { new URI(gcsUri) } catch { case _: Throwable => throw JupyterExtensionException(gcsUri) }
```

Comment:
Parsing will be done in workspace-libs and added to Leo in #41 but this PR has a temporary solution in `GoogleBucketUri `

Reply:
Ooh, I think I forgot about this one, #41 just uses the new parser for the init bucket. Cool that it can be used here too.